third-party/qthread/Makefile

```diff
@@ -144,33 +127,17 @@ endif
 ifneq (, $(CHPL_QTHREAD_SCHEDULER))
 SCHEDULER = $(CHPL_QTHREAD_SCHEDULER)
 endif
-CHPL_QTHREAD_CFG_OPTIONS += --with-scheduler=$(SCHEDULER)
+CHPL_QTHREAD_CFG_OPTIONS += -DQTHREADS_SCHEDULER=$(SCHEDULER)
 
 # only do this on M1+ Macs
 ifeq ($(CHPL_MAKE_TARGET_ARCH),arm64)
 ifeq ($(CHPL_MAKE_TARGET_PLATFORM),darwin)
-  CHPL_QTHREAD_CFG_OPTIONS += --with-cacheline-width=128
+  CHPL_QTHREAD_CFG_OPTIONS += -DQTHREADS_CACHELINE_SIZE_ESTIMATE=128
 endif
 endif
 
 # reduce performance penalty in cases where numChapelTasks < numQthreadWorkers
-CHPL_QTHREAD_CFG_OPTIONS += --enable-condwait-queue
-
-# pgi doesn't support the proper atomic intrinsics that are required for
-# spinlocks to be fast, so disable them for pgi, and PrgEnv-pgi
-ifneq (, $(findstring pgi,$(CHPL_MAKE_TARGET_COMPILER)))
-CHPL_QTHREAD_CFG_OPTIONS += --disable-internal-spinlock
-endif
-
-# Address sanitizer doesn't work with asm context switching so try
-# makecontext/swapcontext (which may have false-positives, but doesn't
-# immediately fail)
-ifneq (,$(CHPL_QTHREAD_SANITIZER_SUPPORT))
-  CHPL_QTHREAD_CFG_OPTIONS += --disable-fastcontext
-  ifeq ($(CHPL_MAKE_TARGET_ARCH),arm64)
-    CFLAGS += -D_XOPEN_SOURCE
-  endif
```

Comment:
I don't see the equivalent new code for this with CMake

Reply:
Good question. fastcontext should now work fine with the various clang sanitizers. It does within qthreads, though it may be worth checking that that support works throughout chapel too.

Reply:
I confirmed that ASAN works properly (at least on my Mac) with this branch as is, awesome!